Compile fails under arm64

**Describe the bug**
Can't compile under arm64
```
   Compiling wasmtime-jit v0.33.0
error[E0433]: failed to resolve: use of undeclared crate or module `rustix`
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/wasmtime-jit-0.33.0/src/code_memory.rs:60:13
   |
60 |             rustix::process::membarrier(
   |             ^^^^^^ use of undeclared crate or module `rustix`
error[E0433]: failed to resolve: use of undeclared crate or module `rustix`
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/wasmtime-jit-0.33.0/src/code_memory.rs:61:17
   |
61 |                 rustix::process::MembarrierCommand::RegisterPrivateExpeditedSyncCore,
   |                 ^^^^^^ use of undeclared crate or module `rustix`
error[E0433]: failed to resolve: use of undeclared crate or module `rustix`
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/wasmtime-jit-0.33.0/src/code_memory.rs:174:17
    |
174 |                 rustix::process::membarrier(
    |                 ^^^^^^ use of undeclared crate or module `rustix`
error[E0433]: failed to resolve: use of undeclared crate or module `rustix`
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/wasmtime-jit-0.33.0/src/code_memory.rs:175:21
    |
175 |                     rustix::process::MembarrierCommand::PrivateExpeditedSyncCore,
    |                     ^^^^^^ use of undeclared crate or module `rustix`
```

**To Reproduce**
```
FROM ubuntu:18.04 as build

RUN apt-get update -qq && apt-get install -y \
    git \
    cmake \
    g++ \
    pkg-config \
    libssl-dev \
    curl \
    llvm \
    clang \
    unzip \
    && rm -rf /var/lib/apt/lists/*

ENV NEAR_VERSION="1.23.1"
ENV NEAR_COMMIT="9110028c663b5f9c1a353e064a535f0ebedd82e0"

# https://github.com/near/nearcore/archive/refs/tags/1.23.1.tar.gz
# ADD https://github.com/near/nearcore/archive/refs/tags/${NEAR_VERSION}.tar.gz /tmp/nearcore-${NEAR_VERSION}.tar.gz
# RUN tar xf /tmp/nearcore-${NEAR_VERSION}.tar.gz \
#    && mv ./nearcore-${NEAR_VERSION} /tmp/near

ADD https://github.com/near/nearcore/archive/${NEAR_COMMIT}.zip /tmp/nearcore.zip
RUN unzip /tmp/nearcore.zip \
    && mv ./nearcore-${NEAR_COMMIT} /tmp/near

COPY ./install-rust.sh /tmp/install-rust.sh

ENV RUSTUP_HOME="/usr/local/rustup" \
    CARGO_HOME="/usr/local/cargo" \
    PATH="/usr/local/cargo/bin:$PATH"
RUN cp /tmp/near/rust-toolchain.toml /tmp/rust-toolchain.toml \
    && /tmp/install-rust.sh \
    && rm /tmp/install-rust.sh

WORKDIR /tmp/near

ENV PORTABLE=ON

RUN CARGO_TARGET_DIR=/tmp/target cargo build -p neard --release --bin neard \
    && CARGO_TARGET_DIR=/tmp/sandbox cargo build -p neard --features sandbox --release
```
**Expected behavior**
Getting near compiled

**Screenshots**

**Version (please complete the following information):**
`9110028c663b5f9c1a353e064a535f0ebedd82e0` from #6237

- rust version:
1.58.1

--- a/runtime/near-vm-runner/Cargo.toml
+++ b/runtime/near-vm-runner/Cargo.toml
@@ -24,7 +24,7 @@ memoffset = "0.6"
 loupe = "0.1"
 once_cell = "1.5.2"
 parity-wasm = { version = "0.42", default-features = false }
-wasmtime = { version = "0.34.2", default-features = false, features = ["cranelift"], optional = true }
+wasmtime = { version = "0.37.0", default-features = false, features = ["cranelift", "wasm-backtrace"], optional = true }
 anyhow = { version = "1.0.19", optional = true }
 near-cache = { path = "../../utils/near-cache" }
 near-vm-logic = { path = "../near-vm-logic", default-features = false, features = [] }

--- a/runtime/near-vm-runner/src/wasmtime_runner.rs
+++ b/runtime/near-vm-runner/src/wasmtime_runner.rs
@@ -159,7 +159,10 @@ pub(super) fn default_config() -> wasmtime::Config {
     config.wasm_bulk_memory(WASM_FEATURES.bulk_memory);
     config.wasm_multi_value(WASM_FEATURES.multi_value);
     config.wasm_multi_memory(WASM_FEATURES.multi_memory);
-    assert_eq!(WASM_FEATURES.module_linking, false, "wasmtime currently does not support the module-linking feature");
+    assert_eq!(
+        WASM_FEATURES.module_linking, false,
+        "wasmtime currently does not support the module-linking feature"
+    );
     config
 }
 

--- a/runtime/near-vm-runner/src/wasmtime_runner.rs
+++ b/runtime/near-vm-runner/src/wasmtime_runner.rs
@@ -152,13 +152,14 @@ pub fn get_engine(config: &mut wasmtime::Config) -> Engine {
 
 pub(super) fn default_config() -> wasmtime::Config {
     let mut config = wasmtime::Config::default();
+    config.max_wasm_stack(1024 * 1024 * 1024).unwrap(); // wasm stack metering is implemented by pwasm-utils, we don't want wasmtime to trap before that
     config.wasm_threads(WASM_FEATURES.threads);
     config.wasm_reference_types(WASM_FEATURES.reference_types);
     config.wasm_simd(WASM_FEATURES.simd);
     config.wasm_bulk_memory(WASM_FEATURES.bulk_memory);
     config.wasm_multi_value(WASM_FEATURES.multi_value);
     config.wasm_multi_memory(WASM_FEATURES.multi_memory);
-    config.wasm_module_linking(WASM_FEATURES.module_linking);
+    assert_eq!(WASM_FEATURES.module_linking, false, "wasmtime currently does not support the module-linking feature");
     config
 }
 

